test/DXswapFactory.spec.ts

```diff
@@ -1,64 +1,79 @@
 import { factoryFixture } from './shared/fixtures'
+import { DXswapFactory, DXswapFeeSetter, DXswapPair__factory } from './../typechain'
+import { bytecode as dxSwapPairBytecode } from '../build/artifacts/contracts/DXswapPair.sol/DXswapPair.json'
+import { getCreate2Address } from './shared/utilities';
 
-import DXswapPair from '../build/DXswapPair.json'
 
-chai.use(solidity)
+const { AddressZero } = constants
 
 const TEST_ADDRESSES: [string, string] = [
   '0x1000000000000000000000000000000000000000',
   '0x2000000000000000000000000000000000000000'
 ]
 
+const overrides = {
+  gasLimit: 9999999
+}
+
 describe('DXswapFactory', () => {
-  const provider = new MockProvider({
-    hardfork: 'istanbul',
-    mnemonic: 'horn horn horn horn horn horn horn horn horn horn horn horn',
-    gasLimit: 9999999
+  const provider = ethers.provider
+  let dxdao: SignerWithAddress
+  let tokenOwner: SignerWithAddress
+  let ethReceiver: SignerWithAddress
+  let fallbackReceiver: SignerWithAddress
+  let other: SignerWithAddress
+  let randomTestFeeSetter: SignerWithAddress
+  let factory: DXswapFactory
+  let feeSetter: DXswapFeeSetter
+
+  beforeEach('assign wallets', async function () {
+    const signers = await ethers.getSigners()
+    dxdao = signers[0]
+    tokenOwner = signers[1]
+    ethReceiver = signers[2]
+    fallbackReceiver = signers[3]
+    other = signers[4]
+    randomTestFeeSetter = signers[5]
   })
-  const [wallet, protocolFeeReceiver, other] = provider.getWallets()
-  const loadFixture = createFixtureLoader(provider, [wallet, protocolFeeReceiver, other])
-
-  let factory: Contract
-  let feeSetter: Contract
-  beforeEach(async () => {
-    const fixture = await loadFixture(factoryFixture)
-    factory = fixture.factory
+
+  beforeEach('deploy fixture', async () => {
+    const fixture = await factoryFixture(provider, [dxdao, ethReceiver, fallbackReceiver])
+    factory = fixture.dxswapFactory
     feeSetter = fixture.feeSetter
-    
-    // Set feeToSetter to wallet.address to test the factory methdos from an ETH account
-    await feeSetter.setFeeTo(AddressZero);
-    await feeSetter.setFeeToSetter(wallet.address);
+
+    // Set feeToSetter to dxdao.address to test the factory methdos from an ETH account
+    await feeSetter.setFeeTo(AddressZero, overrides);
+    await feeSetter.setFeeToSetter(randomTestFeeSetter.address);
   })
 
   it('feeTo, feeToSetter, allPairsLength, INIT_CODE_PAIR_HASH', async () => {
     expect(await factory.feeTo()).to.eq(AddressZero)
-    expect(await factory.feeToSetter()).to.eq(wallet.address)
+    expect(await factory.feeToSetter()).to.eq(randomTestFeeSetter.address)
     expect(await factory.allPairsLength()).to.eq(0)
-    expect(await factory.INIT_CODE_PAIR_HASH()).to.eq('0xd306a548755b9295ee49cc729e13ca4a45e00199bbd890fa146da43a50571776')
+    expect(await factory.INIT_CODE_PAIR_HASH()).to.eq('0x9e43bdf627764c4a3e3e452d1b558fff8466adc4dc8a900396801d26f4c542f2')
```

Comment:
Unsure if maybe here we'd like to dynamically calculate the init code hash by pulling it from the artifact?? Maybe yes, maybe not, something to think about.

Reply:
Nice catch! Imo it is ok to leave like it is though. Firstly to explicitly show hash, it was clear for me when tests failed what happened. Secondly init code doesn't (and shouldn't) change so often to require dynamic calculations.